Source/spells.cpp

```diff
@@ -293,42 +289,32 @@ void DoResurrect(int pnum, uint16_t rid)
 	}
 }
 
-void DoHealOther(int pnum, uint16_t rid)
+void DoHealOther(const Player &caster, Player &target)
 {
-	if (pnum == MyPlayerId) {
-		NewCursor(CURSOR_HAND);
-	}
-
-	if ((DWORD)pnum >= MAX_PLRS || rid >= MAX_PLRS) {
-		return;
-	}
-	auto &player = Players[pnum];
-	auto &target = Players[rid];
-
 	if ((target._pHitPoints >> 6) <= 0) {
 		return;
 	}
 
 	int hp = (GenerateRnd(10) + 1) << 6;
-	for (int i = 0; i < player._pLevel; i++) {
+	for (int i = 0; i < caster._pLevel; i++) {
 		hp += (GenerateRnd(4) + 1) << 6;
 	}
-	for (int i = 0; i < GetSpellLevel(pnum, SPL_HEALOTHER); i++) {
+	for (int i = 0; i < caster.GetSpellLevel(SPL_HEALOTHER); i++) {
 		hp += (GenerateRnd(6) + 1) << 6;
 	}
 
-	if (player._pClass == HeroClass::Warrior || player._pClass == HeroClass::Barbarian) {
+	if (caster._pClass == HeroClass::Warrior || caster._pClass == HeroClass::Barbarian) {
 		hp *= 2;
-	} else if (player._pClass == HeroClass::Rogue || player._pClass == HeroClass::Bard) {
+	} else if (caster._pClass == HeroClass::Rogue || caster._pClass == HeroClass::Bard) {
 		hp += hp / 2;
-	} else if (player._pClass == HeroClass::Monk) {
+	} else if (caster._pClass == HeroClass::Monk) {
 		hp *= 3;
 	}
 
 	target._pHitPoints = std::min(target._pHitPoints + hp, target._pMaxHP);
 	target._pHPBase = std::min(target._pHPBase + hp, target._pMaxHPBase);
 
-	if (rid == MyPlayerId) {
+	if (std::addressof(target) == MyPlayer) {

```

Comment:
Is there a good reason to start using `std::addressof` here?

Reply:
I actually thought we'd been using this already 😅 . There's no real need given there's no `Player::operator&` overload so either way works.

Reply:
```suggestion
	if (&target == MyPlayer) {
```